README.md

```diff
@@ -151,14 +151,53 @@ Creates the http object. In case of failures, returns `nil` and a string describ
 
 ## connect
 
+`syntax: ok, err = httpc:connect(aio_options_table)`
+
 `syntax: ok, err = httpc:connect(host, port, options_table?)`
 
```

Comment:
Do you think perhaps we should deprecate the old interface? Seems like we could use your new interface, without ssl or proxy options, and get the same behaviour without much cost?

Reply:
yes.

These days everything is ssl mostly. Doing the steps manual is causing a lot of code duplication, and is very tedious because there are so many edge cases one needs to think of. A lot of parameters to each of the `connect`, `handshake`, and `proxy` calls share a lifetime over all the three methods.

That is exactly what caused me to rewrite the whole thing, such that it would be less error prone.